### PR TITLE
drivers: adc: stm32 adc disable causing endless loop

### PR DESCRIPTION
**Pull request by FRASTM**
It appears that after merge  of the https://github.com/zephyrproject-rtos/zephyr/pull/38589,
the tests/drivers/adc/adc_api is looping forever on the nucleo_f091rc, nucleo_wb55rg, nucleo_h743zi target boards.
The commit cd3f45d789d95d82c2e5f550ef6ce50b068a59b3 (_drivers: adc: stm32 disable the ADC before setting the Resolution_) of the PR is reverted but the issue  #38578 must still be fixed, this is done waiting for the ADC to be disabled : **only for the stm32L0x**
--> Errata applies only on stm32G0 soc serie only (despite other ref man)  when writing ADC_CFGR1 register
Setting Oversampling also applies on stm32L5.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Comment by FRASTM** [comment]
requires https://github.com/zephyrproject-rtos/zephyr/issues/38591 to PASS on nucleo_f091rc

**Comment by FRASTM** [comment]
@rlezuo1, this a add-on to fix the issue #38578

File: drivers/adc/adc_stm32.c [hunk 1]
```diff
@@ -444,13 +436,15 @@ static int start_read(const struct device *dev,
 	defined(CONFIG_SOC_SERIES_STM32L5X) || \
 	defined(CONFIG_SOC_SERIES_STM32WBX) || \
 	defined(CONFIG_SOC_SERIES_STM32WLX)
+#ifdef CONFIG_SOC_SERIES_STM32L0X
```
Comment (erwango) [inline drivers/adc/adc_stm32.c:439]: For readability, let's move this out of the big `#if defined(CONFIG_SOC_SERIES_STM32G0X) || \
	defined(CONFIG_SOC_SERIES_STM32G4X) || \
	defined(CONFIG_SOC_SERIES_STM32H7X) || \
	defined(CONFIG_SOC_SERIES_STM32L0X) || \
	defined(CONFIG_SOC_SERIES_STM32L4X) || \
	defined(CONFIG_SOC_SERIES_STM32L5X) || \
	defined(CONFIG_SOC_SERIES_STM32WBX) || \
	defined(CONFIG_SOC_SERIES_STM32WLX)`

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by FRASTM** [comment]
tests/drivers/adc/adc_api/drivers.adc,arm,nucleo_wb55rg,passed,,na,7.675704717636108,0,0                                tests/drivers/adc/adc_api/drivers.adc,arm,nucleo_g071rb,passed,,na,4.26364511184692,0,0                                tests/drivers/adc/adc_api/drivers.adc,arm,nucleo_g474re,passed,,na,4.451345682144165,0,0                                tests/drivers/adc/adc_api/drivers.adc,arm,stm32f3_disco,passed,,na,6.906596660614014,0,0                                
tests/drivers/adc/adc_api/drivers.adc,arm,nucleo_f429zi,passed,,na,6.7495903968811035,0,0
tests/drivers/adc/adc_api/drivers.adc,arm,nucleo_h743zi,passed,,na,4.004194736480713,0,0
tests/drivers/adc/adc_api/drivers.adc,arm,nucleo_f207zg,passed,,na,7.629014015197754,0,0
tests/drivers/adc/adc_api/drivers.adc,arm,nucleo_f103rb,passed,,na,8.234832763671875,0,0
tests/drivers/adc/adc_api/drivers.adc,arm,disco_l475_iot1,passed,,na,7.938905477523804,0,0
tests/drivers/adc/adc_api/drivers.adc,arm,nucleo_l073rz,passed,,na,21.37311863899231,0,0

**Review by erwango — APPROVED** [review]

**Review by ABOSTM — APPROVED** [review]

**Pull request merged**